Google.Api.Gax/TaskExtensions.cs

```diff
@@ -49,7 +50,98 @@ public static void WaitWithUnwrappedExceptions(this Task task)
                 // Using ExceptionDispatchInfo to keep the original exception stack trace.
                 ExceptionDispatchInfo.Capture(e.InnerExceptions.FirstOrDefault() ?? e).Throw();
             }
+        }
+
+        /// <summary>
+        /// Synchronously waits for the given task to complete.
+        /// Any <see cref="AggregateException"/> thrown is unwrapped to the first inner exception.
+        /// </summary>
+        /// <param name="task">The task to wait for.</param>
+        /// <param name="timeout">A TimeSpan that represents the number of milliseconds to wait, or
+        /// -1 milliseconds to wait indefinitely.</param>
+        public static void WaitWithUnwrappedExceptions(this Task task, TimeSpan timeout)
```

Comment:
name wise, I would prefer `WaitAndUnwrapExceptions`, but not set on it.

Reply:
We've already got the original WaitWithUnwrappedExceptions method, so we can't really change that now.